pytorch_lightning/utilities/data.py

```diff
@@ -121,7 +121,7 @@ def has_len_all_ranks(
     infinite dataloader."""
     try:
         local_length = len(dataloader)
-        total_length = training_type.reduce(torch.tensor(local_length).to(model.device), reduce_op="sum")
+        total_length = training_type.reduce(torch.tensor(local_length).to(training_type.root_device), reduce_op="sum")
```

Comment:
Unrelated to the purpose of this PR but we could create this in the device directly

```suggestion
        total_length = training_type.reduce(torch.tensor(local_length, device=training_type.root_device), reduce_op="sum")
```